src/transformers/modeling_utils.py

```diff
@@ -1244,6 +1244,7 @@ def _autoset_attn_implementation(
         # Here we use config._attn_implementation_internal to check whether the attention implementation was explicitely set by the user.
         # The property `PretrainedConfig._attn_implementation` is never `None`, for backward compatibility (always fall back on "eager").
         # The `hasattr` here is used as some Transformers tests for some reason do not call PretrainedConfig __init__ (e.g. test_no_super_init_config_and_model)
+        requested_attn_implementation = None
```

Comment:
Should this be "default" instead?

Reply:
No, the idea here is to check whether the user passed `attn_implementation="eager"`, `attn_implementation="sdpa"` or `attn_implementation="sdpa"` explicitly when loading the model from `from_pretrained` or `from_config`.